### PR TITLE
add narrative, expand a bit, minor rewrites

### LINKED ISSUE #6: Add framing narrative
This lesson needs an introductory frame. It's great to jump into the sandwiches example, but students at this point might need to be reminded why objects are great, how we've used them in the past, etc.

### PR DESCRIPTION
**Pull request by scottcreynolds**
resolves #6

### CODE REVIEW

**Comment by scottcreynolds** [comment]
@AnnJohn @pletcher not sure if you guys still want me to tag you in these or just assign to @pletcher  or what lmk!

File: README.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ## Intro
 
+In the [JavaScript Objects](https://learn.co/lessons/javascript-objects) lesson, we learned that an `Object` is the basic associative data structure in JavaScript. We can use these objects to associate data *values* with unique *keys*, giving us a human-readable representation of a logical collection of data.
```
Comment (pletcher) [inline README.md]: I probably should have mentioned this earlier, but one initiative that I've been trying to head up is modularizing our curriculum a bit more. Could we introduce this content without referencing the previous lesson? Totally okay to repeat ourselves by way of review, just as long as this lesson could stand on its own in some other curriculum somewhere.

File: README.md [hunk 2]
```diff
@@ -41,47 +43,88 @@ var grilledCheese = {
 
 ```
 
-Ok but suddenly this got tedious and I only made three sandwiches. You notice that we're copying and pasting a lot of code. Wouldn't it be nice to have one central place to scaffold out what we're building? Using a class in Ruby makes things so much more easy. JavaScript better have something similar...
+Great. Three sandwiches. Plenty to share so we don't get into a...
+situation.
+
+![you ate my sandiwch](http://i.giphy.com/BeurjzzpsYwqQ.gif)
```
Comment (pletcher) [inline README.md:49]: A+

File: README.md [hunk 3]
```diff
@@ -41,47 +43,88 @@ var grilledCheese = {
...
+that are attributes of the object, such as `meat` and `condiments` in a
+`sandwich`), but also allow us to *reuse* the data structure without
+constantly redefining it.
+
+In other words, we should only have to define they properties of a sandwich
+one time and then be able to create as many different sandwiches as we
+want without repeating ourselves.
 
-## Constructor Function
+In Ruby, we'd accomplish this by way of the Ruby `Class`, which we could
```
Comment (pletcher) [inline README.md]: Could we remove the Ruby bits? Students will be a bit blind-sided :)

File: README.md [hunk 4]
```diff
@@ -41,47 +43,88 @@ var grilledCheese = {
...
 
 Next, we define the function to accept a whole bunch of parameters. When we create objects with this constructor function, we'll pass in the value of the properties we want our object to have.
 
-You'll also notice inside the body of the constructor function we're using `this`. In this case, `this` will refer to the current object being created. For now it's ok to think about `this` being similar to Ruby's self.  In this context, `this` refers to the object itself just like `self` does in Ruby.  
-
+You'll also notice inside the body of the constructor function we're using `this`. In this case, `this` will refer to the current object being created. For now it's ok to think about `this` being similar to Ruby's self.  In this context, `this` refers to the object itself just like `self` does in Ruby. We'll explore `this` more in-depth in an an upcoming lesson.
```
Comment (pletcher) [inline README.md]: Hm, we'll have to take out the Ruby stuff

File: README.md [hunk 5]
```diff
@@ -41,47 +43,88 @@ var grilledCheese = {
...
 ```js
-var blt = new Sandwich("white", false, "bacon", "mayo", ["lettuce", "tomato"], "none")
+var blt = new Sandwich("white", false, "bacon", "mayo", ["lettuce", "tomato"], "none");
+
+var turkeyClub = new Sandwich("sourdough", true, ["turkey", "bacon"], "mayo", ["lettuce", "tomato"], "cheddar");
+
+var grilledCheese = new Sandwich("white", false, "none", "none", "none", "cheddar");
+```
+
+Notice that when we call these functions, we always call them with the `new` keyword.  In Ruby, you're used to calling `MyClass.new` to instantiate a new object instance, and in JavaScript we're going to do something very similar.
```
Comment (pletcher) [inline README.md]: :fire: Ruby :fire:

**Comment by pletcher** [comment]
@scottcreynolds thanks for this! Totally cool just to assign to me or to tag us both :)

File: README.md (hunk 1 above)
Comment (scottcreynolds) [inline README.md]: Got it

**Comment by pletcher** [comment]
Awesome sauce -- thanks!

**Pull request merged**